### PR TITLE
BC-8348 - Implement Sharing and Copying of Room Boards

### PR DESCRIPTION
**Pull request by uidp**
# Short Description
We have to implement sharing and copying of boards that are associated to rooms. Currently this is only possible for boards that belong to courses. In order to do this we have to extend the ShareTokenModule and BoardModule.

## Links to Ticket and related Pull-Requests
BC-8348

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/locales/es.ts [hunk 1]
```diff
@@ -597,6 +597,10 @@ export default {
 	"components.molecules.import.columnBoard.options.selectCourse.infoText":
 		"Seleccione el curso al que desea importar el tablero.",
 	"components.molecules.import.courses.label": "Curso",
+	"components.molecules.import.columnBoard.options.selectRoom":
+		"Elija habitación",
```
Comment (odalys-dataport) [inline src/locales/es.ts:601]: We decided on going with "sala" instead of "habitación"

File: src/modules/page/room/RoomDetails.page.vue [hunk 2]
```diff
@@ -49,12 +52,13 @@
...
 
 							<VListItem
+								v-if="canDeleteBoard"
 								role="menuitem"
 								data-testid="room-action-delete"
 								:aria-label="
 									$t('pages.roomDetails.ariaLabels.menu.action.delete')
 								"
-								@click="onDelete"
+								@click="deleteBoard"
```
Comment (odalys-dataport) [inline src/modules/page/room/RoomDetails.page.vue:61]: A while ago we decided that "on" + eventName is our convention to name event handlers

**Review by odalys-dataport — CHANGES_REQUESTED** [review]

File: src/locales/es.ts (hunk 1 above)
Comment (uidp) [reply src/locales/es.ts:601]: Ok, can you help with the right translation? What would it be then?

File: src/modules/page/room/RoomDetails.page.vue (hunk 2 above)
Comment (uidp) [reply src/modules/page/room/RoomDetails.page.vue:73]: Ok, but it reads much better this way IMHO (like "At click: delete board"). And what about methods that are called from event handlers + from other methods? Should we wrap these methods in `on<method>` functions so they can be called in event handlers? Example would be `createBoard` which is called from `fabItemClickHandler`.

File: src/modules/data/room/roomMembers/roomMembers.composable.unit.ts [hunk 3]
```diff
@@ -141,7 +141,7 @@ describe("useRoomMembers", () => {
 					...user,
 					userId: user.id,
 					fullName: `${user.lastName}, ${user.firstName}`,
-					roleName: RoleName.RoomEditor,
+					roleName: RoleName.Roomeditor,
```
Comment (odalys-dataport) [inline src/modules/data/room/roomMembers/roomMembers.composable.unit.ts:144]: Shouldn't it be `RoomEditor`?

File: src/modules/feature/room/RoomMembers/AddMembers.unit.ts [hunk 4]
```diff
@@ -125,11 +125,11 @@ describe("AddMembers", () => {
 			});
 
 			expect(roleComponent).toBeTruthy();
-			await roleComponent.vm.$emit("update:modelValue", RoleName.RoomViewer);
+			await roleComponent.vm.$emit("update:modelValue", RoleName.Roomviewer);
```
Comment (odalys-dataport) [inline src/modules/feature/room/RoomMembers/AddMembers.unit.ts:128]: Same here with `RoomViewer`, maybe you can explain

File: src/modules/data/room/roomMembers/roomMembers.composable.unit.ts (hunk 3 above)
Comment (uidp) [reply src/modules/data/room/roomMembers/roomMembers.composable.unit.ts:144]: That name comes from the API Spec and results from the "specific" naming in the Server. This comes from generated code.

File: src/modules/feature/room/RoomMembers/AddMembers.unit.ts (hunk 4 above)
Comment (uidp) [reply src/modules/feature/room/RoomMembers/AddMembers.unit.ts:128]: see above

File: src/locales/es.ts [hunk 5]
```diff
@@ -597,6 +597,9 @@ export default {
 	"components.molecules.import.columnBoard.options.selectCourse.infoText":
 		"Seleccione el curso al que desea importar el tablero.",
 	"components.molecules.import.courses.label": "Curso",
+	"components.molecules.import.columnBoard.options.selectRoom": "Elija sala",
+	"components.molecules.import.columnBoard.options.selectRoom.infoText":
+		"Seleccione la habitación al que desea importar el tablero.",
```
Comment (MartinSchuhmacher) [inline src/locales/es.ts:602]: Seleccione la sala en la que desea importar el tablero.

**Review by odalys-dataport — DISMISSED** [review]

File: src/modules/page/room/RoomDetails.page.unit.ts [hunk 6]
```diff
@@ -13,19 +13,29 @@ import { RoomVariant, useRoomDetailsStore } from "@data-room";
...
 	useRoute: jest.fn(),
 	useRouter: jest.fn(),
 }));
 
+jest.mock("@feature-room/roomAuthorization.composable");
+const roomPermissions: ReturnType<typeof useRoomAuthorization> = {
+	canCreateRoom: ref(false),
+	canViewRoom: ref(false),
+	canEditRoom: ref(false),
+	canDeleteRoom: ref(false),
```
[the remark this answers is not in the document]
Comment (uidp) [reply src/modules/page/room/RoomDetails.page.unit.ts:35]: Well, can be distinct actions, right?

File: src/modules/data/board/BoardPageInformation.composable.unit.ts [hunk 7]
```diff
@@ -33,78 +33,146 @@ describe("BoardPageInformation.composable", () => {
...
 		});
 	});
 
 	describe("when board context does not exist", () => {
-		beforeEach(() => {
-			jest.resetAllMocks();
-		});
-
-		const setup = () => {
+		const setup = async () => {
```
[the remark this answers is not in the document]
Comment (uidp) [reply src/modules/data/board/BoardPageInformation.composable.unit.ts:175]: removed.

File: src/locales/es.ts (hunk 5 above)
Comment (uidp) [reply src/locales/es.ts:602]: Fixed.

**Review by MartinSchuhmacher — APPROVED** [review]

**Pull request closed without merging**